src/drivers/driver.ts

```diff
@@ -882,7 +882,22 @@ export abstract class CMakeDriver implements vscode.Disposable {
     const compilerDir = path.parse(compilerPath).dir;
 
     // Find an equivalent in the compilers allowed list.
-    const compiler = this.compilerAllowList.find(comp => compilerName.includes(comp.name));
+    // To avoid finding "cl" instead of "clang" or "g++" instead of "clang++",
+    // sort the array from lengthier to shorter, so that the find operation
+    // would return the most precise match.
+    // The find condition must be "includes" instead of "equals"
+    // (which wouldn't otherwise need the sort) to avoid implementing separate handling
+    // for comiler file name prefixes and suffixes related to targeted architecture.
+    const sortedCompilerAllowList = this.compilerAllowList.sort((a, b) => {
+      if (a.name.length == b.name.length) {
```

Comment:
If we're only comparing string lengths, you could just do `return b.name.length - a.name.length;`  (though test and make sure I didn't put `a` and `b` in the wrong spot 😄)

Reply:
Yes great idea. I just submitted. I also realized a test would be helpful and I created one.